engine/table/src/main/java/io/deephaven/stream/TablePublisherImpl.java

```diff
@@ -0,0 +1,172 @@
+package io.deephaven.stream;
+
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.context.ExecutionContext;
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.rowset.RowSequence.Iterator;
+import io.deephaven.engine.rowset.RowSet;
+import io.deephaven.engine.table.ChunkSource.FillContext;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.engine.table.SharedContext;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.engine.table.impl.NotificationStepSource;
+import io.deephaven.engine.table.impl.remote.ConstructSnapshot;
+import io.deephaven.engine.table.impl.remote.ConstructSnapshot.SnapshotFunction;
+import io.deephaven.engine.table.impl.remote.ConstructSnapshot.State;
+import io.deephaven.engine.table.impl.sources.ReinterpretUtils;
+import io.deephaven.util.SafeCloseable;
+import io.deephaven.util.SafeCloseableArray;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+class TablePublisherImpl implements StreamPublisher {
+
+    private final String name;
+    private final TableDefinition definition;
+    private final int chunkSize;
+    private StreamConsumer consumer;
+
+    TablePublisherImpl(String name, TableDefinition definition, int chunkSize) {
+        if (chunkSize <= 0) {
+            throw new IllegalArgumentException("chunkSize must be positive");
+        }
+        this.name = Objects.requireNonNull(name);
+        this.definition = Objects.requireNonNull(definition);
+        this.chunkSize = chunkSize;
+    }
+
+    @Override
+    public void register(@NotNull StreamConsumer consumer) {
+        if (this.consumer != null) {
+            throw new IllegalStateException("Can not register multiple StreamConsumers.");
+        }
+        this.consumer = Objects.requireNonNull(consumer);
+    }
+
+    public void add(Table table) {
+        definition.checkMutualCompatibility(table.getDefinition());
```

Comment:
This is doing a fair bit of extra work. We just care that `table` has a super-set of `definition's` columns, with matching names, data types, and component types.

We agreed that the right answer is probably to dump the test.